stl/inc/iterator

```diff
@@ -994,22 +996,24 @@ public:
 
     template <indirectly_swappable<_Iter> _OIter, class _OSe>
     friend void iter_swap(const common_iterator& _Left, const common_iterator<_OIter, _OSe>& _Right) noexcept(
-        noexcept(_RANGES iter_swap(_Left._Val._Iterator, _Right._Val._Iterator))) {
+        noexcept(_RANGES iter_swap(_Left._Val._Iterator, _Right._Get_val()._Iterator))) {
+        auto& _Right_val = _Right._Get_val();
 #if _ITERATOR_DEBUG_LEVEL != 0
         _STL_VERIFY(_Left._Val._Contains == _Variantish_state::_Holds_iter
-                        && _Right._Val._Contains == _Variantish_state::_Holds_iter,
+                        && _Right_val._Contains == _Variantish_state::_Holds_iter,
             "can only iter_swap common_iterators if both hold iterators");
 #endif // _ITERATOR_DEBUG_LEVEL != 0
-        return _RANGES iter_swap(_Left._Val._Iterator, _Right._Val._Iterator);
+        return _RANGES iter_swap(_Left._Val._Iterator, _Right_val._Iterator);
     }
 
-private:
-    // clang-format off
-    template <input_or_output_iterator _OIter, sentinel_for<_OIter> _OSe>
-        requires (!same_as<_OIter, _OSe> && copyable<_OIter>)
-    friend class common_iterator;
-    // clang-format on

```

Comment:
Not the removal of cross-specialization friendship here; I've chosen to use `_Get_val` consistently in both member functions and friend functions.